sentry_sdk/crons/decorator.py

```diff
@@ -1,19 +1,19 @@
-from functools import wraps
 import sys
 
 from sentry_sdk._compat import reraise
 from sentry_sdk._types import TYPE_CHECKING
 from sentry_sdk.crons import capture_checkin
 from sentry_sdk.crons.consts import MonitorStatus
 from sentry_sdk.utils import now
-
+from sentry_sdk._compat import contextmanager
 
 if TYPE_CHECKING:
-    from typing import Any, Callable, Optional
+    from typing import Generator, Optional
 
 
+@contextmanager
 def monitor(monitor_slug=None):
-    # type: (Optional[str]) -> Callable[..., Any]
+    # type: (Optional[str]) -> Generator[None, None, None]
     """
     Decorator to capture checkin events for a monitor.
```

Comment:
We should probably update the docstring here to also mention the possibility of using `monitor` as a context manager now. An example would be great, too.

Reply:
I actually already had updated the docstring locally, but I forgot to commit and push the change. Thank you for catching this; please let me know what you think about the updated docstring!

Reply:
Looking good! Just a couple of nitpicks:

- I'd write "context manager" instead of "contextmanager"; "contextmanager" is just the name of the decorator that turns functions into context managers while "context manager" is what the concept is called.
- I'd move the Celery disclaimer ("This does not have to be used with Celery, but if you do use it with celery, put the `@sentry_sdk.monitor` decorator below Celery's `@app.task` decorator.") under the decorator example since it's only relevant for the decorator usecase.

Reply:
Okay, will make these changes then merge